### PR TITLE
docs: update transifex links

### PR DESCRIPTION
**Pull request by marcuswin**
Minor link fixes

### CODE REVIEW

**Comment by fanquake** [comment]
Please squash into a single commit, and fix up the commit message.

File: doc/developer-notes.md [hunk 1]
```diff
@@ -185,7 +185,7 @@ Not OK (used plenty in the current source, but not picked up):
 //
 ```
 
-A full list of comment syntaxes picked up by Doxygen can be found at https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html,
+A full list of comment syntaxes picked up by Doxygen can be found at http://www.doxygen.nl/manual/docblocks.html,
```
Comment (laanwj) [inline doc/developer-notes.md]: whoa this site really doesn't support `https`

**Comment by laanwj** [comment]
It looks like both the old and new transifex link works fine. Do you have a reference that the new link format is preferred and the old one might go away at some point?

(you missed one in `doc/release-notes.md`, by the way)

**Comment by fanquake** [comment]
I've finished this up in #15514.

**Pull request closed without merging**